API.md

```diff
@@ -0,0 +1,229 @@
+# Pax API documentation
+An overview of programming for Pax is given below, followed by a [description of
+the API](#Details).
+
+## Overview
+
+## Programming model
+Using Pax you describe a network packet processor in terms of what it does to
+packets sent and received through network logical *ports*. At runtime, these
+ports are attached by Pax to network interfaces made available by your OS. The
+interfaces may be physical or virtual (e.g., a tap device).
+
+![A Pax packet processor](http://www.cl.cam.ac.uk/~ns441/pax/packetproc.png)
+
+A Pax processor can have any number of ports (numbered 0-3 in the drawing
+above) which serve as the main interface with the outside world. The processor
+can write to any of these ports, and processes data that arrives on some of the
+ports (according to its configuration).
+
+
+# What does a Pax processor look like?
+The main handler function of our [NAT example](examples/Nat/NATBase.cs) looks like this.
+The return value is the port over which to emit the (modified) packet. The
+actual network interface connected to that port is determined by the
+configuration file.
+You can also read more about Pax's [API](API.md).
```

Comment:
You link here to the API page from the API page. Did you intend to link to a section? Or perhaps it was an accidental copy from the main README?